### ISSUE TITLE
NPE in _ElementsFlowViewState

### ISSUE THREAD

**Issue by bakua** [MasahideMori-SimpleAppli/sakura_blizzard#3]
Hello, first of all, thank you for the library. Makes the app super pretty.

 I noticed the following NPE in Crashlytics.

Null check operator used on a null value. Error thrown Null check operator used on a null value.
       at State.setState(framework.dart:1223)
       at _ElementsFlowViewState._loadImage.<fn>(elements_flow_view.dart:126)

This is my code:

```
  List<Uint8List> images = [];

  @override
  void initState() {
    super.initState();
    () async {
      Future<Uint8List> readFileBytes(String filePath) async {
        final bd = await rootBundle.load(filePath);
        return bd.buffer.asUint8List(bd.offsetInBytes, bd.lengthInBytes);
      }

      images = [
        await readFileBytes(Assets.images.snowflake1.path),
        await readFileBytes(Assets.images.snowflake2.path),
        await readFileBytes(Assets.images.snowflake3.path),
      ];

      setState(() {});
    }();
  }

  @override
  Widget build(BuildContext context) {
...
if (images.isNotEmpty)
                    Opacity(
                      opacity: .1,
                      child: ImageFallView(
                        viewSize: Size(constraints.maxWidth, constraints.maxHeight),
                        fps: 60,
                        images: images,
                        backImages: images,
                        dropType: EnumDropType.hirahiraDrop,
                        child: const SizedBox.shrink(),
                      ),
                    ),
```

**Comment by MasahideMori-SimpleAppli** [comment]
Hi @bakua ,
Thank you open this issue.

Thank you for trying this package!

I tried similar code on my side, but no error occurred. Under normal circumstances, it seems unlikely that "elements_flow_view.dart:126" would cause a null error.
When this error occurs, I think the following cases are likely, but I don't have enough information to judge.

1. When a specific bit becomes an invalid value due to a problem with the physical device or OS, such as a smartphone. This can occur due to equipment failure or a side effect of the miniaturization of circuits in modern smartphones.
2. If the corresponding resource does not exist or is corrupted, or if acquisition fails due to an access error or permission error. However, I think this usually results in other types of errors.
3. If the executable file is corrupted for some reason during compilation.
(for example, if you make a mistake in writing a specific import in a Flutter app, the generated program will be corrupted in a complicated way [https://github.com/dart- lang/linter/issues/3244]).

Does the error you reported always occur?
If the error occurs only in rare cases, it is likely that this package is not the cause.

**Comment by MasahideMori-SimpleAppli** [comment]
Hi @bakua ,

Sorry. I forgot to consider one possible case.
I currently believe that the error was caused by the ImageFallView widget being destroyed fast for some reason.
I plan to release a revised version in the near future. I will notify you here when a new version is available.

Thank you!

**Comment by bakua** [comment]
Hi, thank you for letting me know :) Yes, I don't think that any of the first three possibilities is the case as the error occurs often enough. My first hunch would be some lifecycle bug indeed.

Good luck hunting it down and let me know if I can help somehow :)

**Comment by MasahideMori-SimpleAppli** [comment]
Hi @bakua,

I have released a modified version 3.1.4.

Thanks to you opening this issue, this package is now more stable.
I'll close this issue for now, but if you still encounter the error, please let me know in the comments.

Thank you!
